app/piping/plumbing_preprocessor.py

```diff
@@ -12,13 +11,51 @@
 logger = logging.getLogger(__name__)
 
 
+def get_schema_template_context(questionnaire_manager, schema):
+    '''
+    :param questionnaire_manager: to retrieve answers
+    :param schema:
+    :return: context dict for schema variable substitution.
+    '''
+    return {
+      "exercise": _build_exercise(),
+      "answers": _build_answers(questionnaire_manager, schema),
+    }
+
+
+def _build_exercise():
+    '''
+    Build the exercise data from the survey metadata
+    '''
+    metadata = get_metadata(current_user)
+
+    return {
+        "start_date": to_date(metadata["ref_p_start_date"]),
+        "end_date": to_date(metadata["ref_p_end_date"]),
+        "employment_date": to_date(metadata["employment_date"]),
+        "return_by": to_date(metadata["return_by"]),
+    }
+
+
+def _build_answers(questionnaire_manager, schema):
+    '''
+    Get the answer values for all aliased elements and make them available for piping.
+    Where answers are not available, use an empty string
+    '''
+    aliases = schema.aliases
+    values = {}
+    for alias, item_id in aliases.items():
+        value = questionnaire_manager.find_answer(item_id)
+        if value is None:
+            value = ""  # Empty string
+        values[alias] = value
+    return values
+
+
 class PlumbingPreprocessor(object):
 
     def plumb_current_state(self, questionnaire_manager, state, schema):
```

Comment:
I started doing this and reverted it because I thought the PR was big enough as it is. Maybe we can visit as a part of the removing of the plumbing stuff; it will probably naturally fall away?